### PR TITLE
A Tiny JS World

### PR DESCRIPTION
**Pull request by SofiiaTrokhymchuk**
### A Tiny JS World

[Demo](https://sofiiatrokhymchuk.github.io/a-tiny-JS-world/) |
[Code base](https://github.com/SofiiaTrokhymchuk/a-tiny-JS-world)

The code is submitted in a dedicated feature branch.

Only code files are submitted.

Please, review.

### CODE REVIEW

File: submissions/SofiiaTrokhymchuk/tiny-js-world/index.js [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+catWoman.legs = 2;
+catWoman.hands = 2;
+
+function inhabitantToString(inhabitant){
+   const friends = [];
+   if(inhabitant.friends === undefined){
+      friends.push(0);
+   }else{
+      friends.push(...inhabitant.friends);
+   }
```
Comment (OleksiyRudenko) [inline submissions/SofiiaTrokhymchuk/tiny-js-world/index.js:58]: 1) Can we employ ternary operator here?
2) `friends.push(0)`: why not assign a value of `[0]`?
3) `friends.push(...inhabitant.friends)`: `arr = [...sourceArr]` is another (ES6) way to copy elements into a new array. But why do we need to have a copy at all?

**Review by OleksiyRudenko — CHANGES_REQUESTED** [review]
@SofiiaTrokhymchuk well done!
A couple of improvements and we're good to go.

File: submissions/SofiiaTrokhymchuk/tiny-js-world/index.js [hunk 2]
```diff
@@ -50,12 +50,8 @@ catWoman.legs = 2;
...
 
 function inhabitantToString(inhabitant){
-   const friends = [];
-   if(inhabitant.friends === undefined){
-      friends.push(0);
-   }else{
-      friends.push(...inhabitant.friends);
-   }
+   let friends = [];
+   inhabitant.friends === undefined ? friends[0] = 0 : friends = inhabitant.friends;
```
Comment (OleksiyRudenko) [inline submissions/SofiiaTrokhymchuk/tiny-js-world/index.js:54]: 1. Proper use of [Ternary operator](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Conditional_Operator)
2. `[0]` is a legitimate assignable value.
3. These 2 statements can be a single statements.

**Review by OleksiyRudenko — CHANGES_REQUESTED** [review]
@SofiiaTrokhymchuk a good optimization step.
Let's make yet another one to have even less verbose code.

File: submissions/SofiiaTrokhymchuk/tiny-js-world/index.js (hunk 2 above)
Comment (SofiiaTrokhymchuk) [reply submissions/SofiiaTrokhymchuk/tiny-js-world/index.js:54]: You mean, that I need to combine a declaration and initialisation of `friends` and ternary operator into a single statement?

Comment (OleksiyRudenko) [reply submissions/SofiiaTrokhymchuk/tiny-js-world/index.js:54]: Yes

**Review by OleksiyRudenko — CHANGES_REQUESTED** [review]
@SofiiaTrokhymchuk nice try. Although it went sideways.

File: submissions/SofiiaTrokhymchuk/tiny-js-world/index.js [hunk 3]
```diff
@@ -50,10 +50,9 @@ catWoman.legs = 2;
 catWoman.hands = 2;
 
 function inhabitantToString(inhabitant){
-   let friends = [];
-   inhabitant.friends === undefined ? friends[0] = 0 : friends = inhabitant.friends;
+   inhabitant.friends === undefined ? inhabitant.friends = [0] : inhabitant.friends = inhabitant.friends;
```
[the remark this answers is not in the document]
Comment (SofiiaTrokhymchuk) [reply submissions/SofiiaTrokhymchuk/tiny-js-world/index.js:53]: Sorry, I didn't pay my attention enough. 5 minutes and it will be fixed!

**Review by OleksiyRudenko — APPROVED** [review]
@SofiiaTrokhymchuk great job!

**Pull request merged**